modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/utils/string.spec.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+
+import {toCamelCase} from '../../utils/string';
+
+describe('toCamelCase(str, removeSpecialCharacters, keepFirstLetterCase)', () => {
+	it('returns first letter in uppercase', () => {
+		expect(toCamelCase('FirstLetterUpperCase', false, true)).toBe('FirstLetterUpperCase');
+	});
+
+	it('returns first letter in lowercase', () => {
+		expect(toCamelCase('firstLetterLowerCase', false, true)).toBe('firstLetterLowerCase');
+	});
+
+	it('removes special characters', () => {
+		expect(toCamelCase('specialCharacter$', true, false)).toBe('specialCharacter');
+	});
+
+	it('makes UPPER cASE string CamelCase', () => {
```

Comment:
This test proves that the function does not always do what it says. There are cases where it does not return a camel case. If this were test-driven development, we would run this test locally and modify the toCamelCase until it passed, but because we are already committing, we should remove this and possibly open a bug. For now, only remove this case.